### PR TITLE
[TRAFWKPR-8] Add NetzkartePointLayer

### PR DESCRIPTION
**Pull request by gevos**

### CODE REVIEW

**Comment by gevos** [comment]
> The gjc does not make a lot of sense when using the bbox.

I have replaced by the geoserverUrl

**Comment by gevos** [comment]
@mhaertwig f.y.i.: hideInLegend is not yet implemented

File: src/components/TopicLoader/TopicLoader.js [hunk 1]
```diff
@@ -70,6 +70,10 @@ class TopicLoader extends Component {
   onClick(features, layer, event) {
     const { dispatchSetClickedFeatureInfo } = this.props;
 
+    if (!layer.getVisible() || !features.length) {
```
Comment (mhaertwig) [inline src/components/TopicLoader/TopicLoader.js]: Please remove `state.clickedFeatureInfo` is an empty array, it means no features are clicked.
Moreover, invisible can't be clicked which is why checking the visibility of the layer is not necessary here.

File: src/layers/NetzkartePointLayer/NetzkartePointLayer.js [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    let key = 'ch.sbb.stationen';
+
+    if (options.showAirports) {
+      name = 'Flughäfen';
+      key = 'ch.sbb.flughafen';
+    }
+
+    const vectorSource = new OLVectorSource({
+      format: new OLGeoJSON(),
+      ...(options.useBboxStrategy && { strategy: OLBboxStrategy }),
```
Comment (mhaertwig) [inline src/layers/NetzkartePointLayer/NetzkartePointLayer.js]: I don't find this a good practice. Please use a ternary operator or a condition.

File: src/layers/NetzkartePointLayer/NetzkartePointLayer.js [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    const olLayer = new OLVectorLayer({
+      source: vectorSource,
+    });
+
+    super({
+      ...options,
+      name,
+      key,
+      olLayer,
+      radioGroup: 'stationen',
```
Comment (mhaertwig) [inline src/layers/NetzkartePointLayer/NetzkartePointLayer.js]: I prefer `stations`

File: src/layers/NetzkartePointLayer/NetzkartePointLayer.js [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+    });
+
+    // Options
+    this.showAirports = !!options.showAirports;
+    this.useBboxStrategy = !!options.useBboxStrategy;
+
+    // Bindings
+    this.airportStyle = this.airportStyle.bind(this);
```
Comment (mhaertwig) [inline src/layers/NetzkartePointLayer/NetzkartePointLayer.js]: That's not necessary if arrow functions are used.

File: src/layers/NetzkartePointLayer/NetzkartePointLayer.js [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+    this.loader = this.loader.bind(this);
+
+    // Style
+    this.netzkarteStyleCache = {};
+    olLayer.setStyle(
+      options.showAirports ? this.airportStyle : this.defaultStyle,
+    );
+
+    // Url
+    this.url = `${CONF.geoserverUrl}?service=WFS&version=1.0.0&request=GetFeature&`;
```
Comment (mhaertwig) [inline src/layers/NetzkartePointLayer/NetzkartePointLayer.js]: `service`, `version` and `request` are urlParams as well.

File: src/layers/NetzkartePointLayer/NetzkartePointLayer.js [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+    this.urlParams = {
+      typeName: options.showAirports
+        ? 'trafimage:netzkarte_airport_point'
+        : 'trafimage:netzkarte_point',
+    };
+
+    // Set loader after binding
+    this.vectorSource = vectorSource;
```
Comment (mhaertwig) [inline src/layers/NetzkartePointLayer/NetzkartePointLayer.js]: I wouldn't use `vectorSource` as a class parameter. You can always do `this.olLayer.getSource()` which i find more explicit.

File: src/layers/NetzkartePointLayer/NetzkartePointLayer.js [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+    vectorSource.setLoader(this.loader);
+  }
+
+  init(map) {
+    super.init(map);
+    this.map = map;
+    const { olLayer } = this;
```
Comment (mhaertwig) [inline src/layers/NetzkartePointLayer/NetzkartePointLayer.js]: nah

File: src/layers/NetzkartePointLayer/NetzkartePointLayer.js [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+
+    // Clear the layer when the resolution changes
+    // as the WFS is resolution dependent
+    this.map.getView().on('change:resolution', () => {
+      olLayer.getSource().clear();
```
Comment (mhaertwig) [inline src/layers/NetzkartePointLayer/NetzkartePointLayer.js]: `this.olLayer...`

**Comment by oterral** [comment]
@gevos I've fixed the build

File: src/components/TopicLoader/TopicLoader.js (hunk 1 above)
Comment (oterral) [reply src/components/TopicLoader/TopicLoader.js]: @gevos do you forgot to make changes for this comment ?

File: src/layers/NetzkartePointLayer/NetzkartePointLayer.js [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+      olLayer,
+      radioGroup: 'stations',
+    });
+
+    // Options
+    this.showAirports = !!options.showAirports;
+    this.useBboxStrategy = !!options.useBboxStrategy;
+
+    // Bindings
+    this.loader = this.loader.bind(this);
```
Comment (mhaertwig) [inline src/layers/NetzkartePointLayer/NetzkartePointLayer.js]: Could't we use arrow functions?

File: src/popups/NetzkartePopup/NetzkartePopup.js [hunk 10]
```diff
@@ -0,0 +1,147 @@
...
+    this.state = {
+      showPlanLinks: false,
+    };
+  }
+
+  render() {
+    const { feature, language, t } = this.props;
+    const { showPlanLinks } = this.state;
+
+    const isAirport = feature.get('layer').indexOf('flug') > 0;
```
Comment (mhaertwig) [inline src/popups/NetzkartePopup/NetzkartePopup.js]: `> -1`

File: src/popups/NetzkartePopup/NetzkartePopup.scss [hunk 11]
```diff
@@ -0,0 +1,36 @@
+.tm-netzkarte-popup {
```
Comment (mhaertwig) [inline src/popups/NetzkartePopup/NetzkartePopup.scss]: please use `wkp` as a prefix in trafimage-maps.

**Comment by mhaertwig** [comment]
> @mhaertwig f.y.i.: hideInLegend is not yet implemented
@gevos please implement `hideInLegend` and give me a pr.

File: src/components/TopicLoader/TopicLoader.js (hunk 1 above)
Comment (mhaertwig) [reply src/components/TopicLoader/TopicLoader.js]: @gevos please update this.

File: src/components/TopicLoader/TopicLoader.js [hunk 12]
```diff
@@ -70,6 +70,10 @@ class TopicLoader extends Component {
   onClick(features, layer, event) {
     const { dispatchSetClickedFeatureInfo } = this.props;
 
+    if (!features.length) {
```
Comment (mhaertwig) [inline src/components/TopicLoader/TopicLoader.js]: please remove this check. an empty array indicates that no features are clicked.

**Pull request merged**